pcdsutils/ext_scripts.py

```diff
@@ -0,0 +1,99 @@
+import logging
+import re
+import socket
+import subprocess
+
+
+logger = logging.getLogger(__name__)
+CNF = '/reg/g/pcds/dist/pds/{0}/scripts/{0}.cnf'
+SCRIPTS = '/reg/g/pcds/engineering_tools/{}/scripts/{}'
+TOOLS = '/reg/g/pcds/dist/pds/tools/{}/{}'
+
+
+def call_script(args, timeout=None, ignore_return_code=False):
+    logger.debug('Calling external script %s with timeout=%s,'
+                 ' ignore_fail=%s', args, timeout, ignore_return_code)
+    try:
+        return subprocess.check_output(args, universal_newlines=True,
+                                       timeout=timeout)
+    except subprocess.CalledProcessError as exc:
+        if ignore_return_code:
+            return exc.output
+        else:
+            logger.debug('CalledProcessError from %s', args, exc_info=True)
+            raise
+    except Exception:
+        logger.debug('Exception raised from %s', args, exc_info=True)
+        raise
+
+
+cache = {}
+
+
+def cache_script(args, timeout=None, ignore_return_code=False):
+    key = ' '.join(args)
+    try:
+        return cache[key]
+    except KeyError:
+        output = call_script(args, timeout=timeout,
+                             ignore_return_code=ignore_return_code)
+        cache[key] = output
+        return output
+
+
+def clear_script_cache():
+    global cache
+    cache = {}
+
+
+def get_hutch_name(timeout=10):
```

Comment:
If you think it's a good idea to refactor now (perhaps, perhaps not 🤷 ),`@functools.lru_cache()` can replace the caching mechanism pretty easily

Reply:
I think we should do a PR that is just the transplant, and then a second PR that makes adjustments

In addition to using lru cache I want to expand docstrings

Reply:
Works for me 👍

Reply:
The other thing I just realized is that doing it in two PRs makes it really easy to review only the changes we're making

Reply:
Good point - I like it!